### ISSUE TITLE
Using pwndbg debugger to report errors in ubuntu with arm architecture  （macbook m1pro）

### ISSUE THREAD

**Issue by Ray-778** [pwndbg/pwndbg#2190]
This Ubuntu is of the x86 architecture.
I can run x86 programs and scripts normally, but I can't debug them with pwndbg.
The error message is: "Exception occurred: Error: Couldn't get registers: Input/output error. (<class 'gdb.error'>)".
<img width="1057" alt="iShot_2024-05-26_17 31 47" src="https://github.com/pwndbg/pwndbg/assets/57139555/a8f16401-2531-49b8-9114-3be40ad6ab11">

**Comment by xsafter** [comment]
I have the same issue

**Comment by 1ikeadragon** [comment]
same issue did you find a resolution? @Ray-778 @xsafter

**Comment by 1ikeadragon** [comment]
https://github.com/1ikeadragon/pwntainer

This is what I'm using for now.

**Comment by disconnect3d** [comment]
@Ray-778 What is the error if u do `set exception-verbose on`?

Also, how do u run the X86-64 Ubuntu machine on the M1 mbp?

**Comment by 1ikeadragon** [comment]
@disconnect3d check the GitHub link I added in my previous comment. You can either use Rosetta or qemu.

**Comment by 1ikeadragon** [comment]
@disconnect3d it's not your pwndbg issue it's an issue with Rosetta2 and docker on Mac.

**Comment by disconnect3d** [comment]
@1ikeadragon can you provide more information? I mean, I have heard about that already a few times, but it would be great to have it documented here.

**Comment by 1ikeadragon** [comment]
Hi @disconnect3d on what do you want more information? I'll be happy to share

**Comment by disconnect3d** [comment]
I mean, how did the OP run GDB? Did they run `gdb ./binary` on M1 and hit `run` and that binary is emulated via Rosetta and that crashes?

Does this crash because Rosetta (assuming here) can't work around ptrace?

It shouldn't matter but is GDB a x86-64 binary here as well? Or is it ARM? Just wondering...

**Comment by 1ikeadragon** [comment]
@disconnect3d  hope this helps! 

By their wording "This Ubuntu is of the x86 architecture.
I can run x86 programs and scripts normally, but I can't debug them with pwndbg." 

I'd assume they were using docker with the flag `--platform=linux/amd64`. They installed and were running pwndbg inside that container.

This doesn't work because Rosetta2 and Docker on mac [doesn't fully support AMD64 debugging](https://github.com/docker/for-mac/issues/6921) in containers.

I tried the exact same thing and the error OP got is the same one I got when using Rosetta2. FWIW if you use QEMU with Docker instead of Rosetta2, you'll get `PTRACE_NOT_IMPLEMENTED` error. 

Both Rosetta2 and QEMU when used in Docker emulate the user-space. So, both of them fail when debugging because it requires kernel level emulation. 

If you check my repo on contaiiner I've provided links to issues and docs and written in gist. Also provided a solution. Rest assured the `bug` label should be removed because this is **not** a pwndbg or gdb or gef bug.
